veps/sig-compute/10-dra-devices/vep.md

```diff
@@ -0,0 +1,760 @@
+- As a user, I would like to use KubeVirt's default driver.
+- As a user, in heterogeneous clusters, i.e. clusters made of nodes with different hardware managed through DRA drivers,
+  I should be able to easily identify what hardware was allocated to the VMI.
+- As a developer, I would like APIs to be extensible so I can develop drivers/webhooks/automation for custom use-cases.
+- As a device-plugin author, I would like to have a well documented way, intuitive way to support devices in KubeVirt.
+
+## Use Cases
+
+### Supported Usecases
+
+1. Devices where the DRA driver sets agreed upon attributes for resources in ResourceSlice.
+    1. PCIe Bus address for pGPU.
+    2. Mediated device uuid for vGPU.
+1. Devices have to either be of the type GPU or HostDevices.
+
+### Unsupported Usecases
+
+1. Devices where the DRA driver does not set the attributes needed to configure libvirt dom XML for the devices will not
+   be supported.
+    1. In the future a standardization could be envisioned for a PCI device where the attributes are set
+       automatically through the DRA framework. When this is achieved, any DRA device should be available in KubeVirt VMs
+2. NetworkDevices or Storage Devices which need a lifecycle of its own will not be supported through HostDevices in alpha
+   release
+
+## Repos
+
+kubevirt/kubevirt
+
+## Design
+
+For allowing users to consume DRA devices, there are two main changes needed:
+
+1. API changes and the plumbing required in KubeVirt to generate the domain xml with the devices.
+2. Driver Implementation to set the required attributes for KubeVirt to use
+
+This design document focuses on part 1 of the problem. We're introducing a new feature gate `DRADevices`.
+All the API changes will be gated behind this feature gate so as not to break existing functionality.
+
+### API Changes
+
+```go
+type VirtualMachineInstanceSpec struct {
+	..
+	..
+	// ResourceClaims defines which ResourceClaims must be allocated
+	// and reserved before the VMI and hence virt-launcher pod is allowed to start. The resources
+	// will be made available to the domain which consume them
+	// by name.
+	//
+	// This is an alpha field and requires enabling the
+	// DynamicResourceAllocation feature gate in kubernetes
+	//  https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/
+	//
+	// This field is immutable.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	ResourceClaims []k8sv1.PodResourceClaim `json:"resourceClaims,omitempty"`
+}
+
+type GPU struct {
+	// Name of the GPU device as exposed by a device plugin
+	Name string                    `json:"name"`
+	DeviceSource DeviceSource      `json:",inline"`
+	VirtualGPUOptions *VGPUOptions `json:"virtualGPUOptions,omitempty"`
+	// If specified, the virtual network interface address and its tag will be provided to the guest via config drive
+	// +optional
+	Tag string `json:"tag,omitempty"`
```

Comment:
Nit: it would be nice to make it obvious what's old and what's new, like:
```suggestion
	..
	// DeviceName string `json:"deviceName"`    <-- removed in favor of DeviceSource
	DeviceSource DeviceSource      `json:",inline"`
	..
```
same below

Reply:
@jean-edouard updated, PTAL and resolve this comment if it makes sense